src/main/kotlin/com/esosa/f5pi_backend/security/filter/JWTAuthenticationFilter.kt

```diff
@@ -58,14 +63,14 @@ class JWTAuthenticationFilter(
 
     private fun HttpServletRequest.getValidAuthHeader(): String =
         getHeader(AUTHORIZATION)
-            ?.takeIf { it.startsWith("Bearer ") }
+            ?.takeIf { it.startsWith(BEARER_PREFIX) }
             ?: throw JWTAuthException(
                 MISSING_HEADER_EXCEPTION_MESSAGE.first,
                 IllegalArgumentException("Authorization header is missing or invalid")
             )
 
     private fun String.extractToken(): String =
-        substringAfter("Bearer ").takeIf { it.isNotBlank() }
+        substringAfter(BEARER_PREFIX).takeIf { it.isNotBlank() }
             ?: throw IllegalStateException("Invalid JWT format")
```

Comment:
Nunca llega a throwear esto, o no? Ya que se llama antes a la validación del header y lanza la otra excepción, no podriamos eliminarla?

Reply:
En realidad la validación es distinta. La primera validación (del auth header) verifica que exista el header de autorización y que comience con "Bearer ". Si esto ocurre, pasa a esta validación, donde se extrae la subcadena que está **luego de** a "Bearer ".

En definitiva, el usuario puede pegarle a algún endpoint protegido con un auth header que contenga simplemente "Bearer " y pasaría la validación del auth header (porque el encabezado sí comienza con "Bearer ") pero no esta que valida la subcadena siguiente.

Coincido en que la validación es un poco pobre, en realidad se podría hacer algo como

```suggestion
private fun String.extractToken(): String =
        substringAfter(BEARER_PREFIX).takeIf { jwtService.hasValidJWTFormat(it) }
            ?: throw IllegalStateException("Invalid JWT format")
```

Pero esto implicaría hacer un `parse` de los claims y la idea es atar la mayor cantidad de cabos antes de que eso suceda, ya que este `parse` se va a hacer de todos modos para extraer el _subject_. Conclusión, el filtro andaría sin esta validación, pero sí es distinta a la del auth header.

Reply:
Comprendo que la excepción es distinta a la del **auth header**, sin embargo te lo decía ya que incluso si le enviamos un **"Authorization: Bearer "** por alguna razón no toma el takeIf y lanza la excepción. No hay forma de llegar a ese throw (o me estoy perdiendo de algo xd)

Reply:
Capaz no nos estamos entendiendo jaja. Te decía que es posible llegar a ese throw con un auth header "Bearer ", que pasaría el `takeIf` de la validación del authHeader pero no el de la validación del token porque no se está trabajando con el header puro, sino con la cadena subsiguiente a "Bearer " (`substringAfter`).

Si probamos algo como esto:
```
val auth = "Bearer "
println(auth.takeIf { it.startsWith("Bearer ") }
    ?: throw JWTAuthException(
        MISSING_HEADER_EXCEPTION_MESSAGE.first,
        IllegalArgumentException("Authorization header is missing or invalid")
    )
)
println(
    auth.substringAfter("Bearer ").takeIf { it.isNotBlank() }
        ?: throw IllegalStateException("Invalid JWT format")
)
```

Se lanzaría la segunda excepción y no la primera. La clave de todo es el `substringAfter`. Es decir, se puede llegar a ese throw. De todos modos, mientras más veo la validación más al pedo me parece.

Reply:
Jajaja bien, te entendí.

Igual mirá, resulta que si haces una petición como ésta:
```console
curl -X GET \
  http://localhost:8080/users/25ccd226-2b73-4931-93cc-b99c7c92b4eb/games \
  -H "Authorization: Bearer "
```

Vos decís que debería pasar la primera validación puesto que comienza con **"Bearer "** pero no es así, no pasa ya que resulta que de alguna forma en algún punto ese espacio se elimina y se termina enviando un `"Authorization: Bearer"`

Así que por eso no se puede llegar a ese throw de manera práctica. Espero que ahora si nos hayamos entendido(? igual dejalo, no vamos a hacer tanto lio por esto.

Reply:
Ah hell nah, no sabía que CURL te trimeaba los headers. En ese caso es como decís vos.